src/averageAges.js

```diff
@@ -1,61 +1,42 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menArray = people.filter((man) => !century
+    ? man.sex === 'm'
+    : man.sex === 'm' && Math.ceil(man.died / 100) === century);
+
+  return averageLifeDuration(menArray);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const motherNames = people.filter((woman) => woman.mother !== null)
+    .map((a) => a.mother);
+
+  const womenArray = people.filter((woman) => !withChildren
+    ? woman.sex === 'f'
+    : motherNames.includes(woman.name)
+  );
+
+  return averageLifeDuration(womenArray);
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenArray = people.filter((child) => !onlyWithSon
+    ? (people.find(mother => child.mother === mother.name))
+    : (people.find(mother => child.mother === mother.name
+      && child.sex === 'm')));
+
+  return (childrenArray.reduce((difference, child) => {
+    return difference + child.born - people.find(mother => (
+      child.mother === mother.name)).born;
```

Comment:
Will be better to create varible for` people.find(mother => child.mother === mother.name)`